framework/include/userobject/ActivateElementUserObject.h

```diff
@@ -0,0 +1,112 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "ElementUserObject.h"
+#include "Function.h"
+#include "NonlinearSystemBase.h"
+#include "AuxiliarySystem.h"
+
+class ActivateElementUserObject;
+
+template <>
+InputParameters validParams<ActivateElementUserObject>();
```

Comment:
This is deprecated and should not be added in new objects. And you're not using the legacy parameters macro in the .C file either, so this _has_ to be removed here.

Reply:
> Good for me. But you need to address @dschwen 's comments.

Thanks @fdkong for the comments! I will work on @dschwen 's suggestions before merging. So excited to see this PR getting close to be merged!